src/types.ts

```diff
@@ -23,13 +24,16 @@ export interface Query {
   /** rest path associated with api and domain */
   apiPath: 'resource' | 'api/views' | 'api/catalog/v1'
 }
-
 export interface AuthOpts {
   // apiToken?: string
   /** appToken userd to make authenticated req */
   appToken?: string
   /** whether to serialize response keys to camelCase */
   keysCamelCased?: boolean
   /** where to use json or csv files */
-  ext?: 'json' | 'csv'
+  ext?: string
+}
+export interface RequestOption {
```

Comment:
Why do we have this req option! What’s the use case?

Reply:
It's the return type for the function that return the options for a request. that function is called in both createRunner and createCsvRunner. The code inside that option was being repeated so i turned it into its own function with a return type